### PR TITLE
fix(lib): keep selection when selectableItems change (#139)

### LINKED ISSUE #139: Selection lost on update if dtsSelectItem is assigned a value
As far as I understand, I can either do:
`<mat-card [dtsSelectItem]>`
or the following:
`<mat-card [dtsSelectItem]="document.id">`

When using the latter however, I have noticed that when the items are updated, the selection is reset.

To reproduce, see the following [stackblitz](https://stackblitz.com/edit/ngx-drag-to-select-pe3w4v?file=src/app/app.component.html)

- In both containers, select Document 1
- Click the update items button
- Notice that the second selection will be cleared. 
- Afterwards it's also no longer possible to select Document 1 by dragging in the second select-container.

Problem is caused here, I think:
https://github.com/d3lm/ngx-drag-to-select/blob/4b12a79d381547ed3ca046d958cb4b1633d535f9/projects/ngx-drag-to-select/src/lib/select-container.component.ts#L347

`selectedItems` will be of type `number[]`, while `newList` is of type `SelectItemDirective[]`.

### PR DESCRIPTION
**Pull request by lemmywrap**
Pull request which fixes #139

### CODE REVIEW

File: projects/ngx-drag-to-select/src/lib/select-container.component.ts [hunk 1]
```diff
@@ -612,7 +613,7 @@ export class SelectContainerComponent implements AfterViewInit, OnDestroy, After
       success = true;
       selectedItems.splice(index, 1);
       this._selectedItems$.next(selectedItems);
-      this.itemDeselected.emit(item.value);
+      this.itemDeselected.emit(value);
```
Comment (d3lm) [inline projects/ngx-drag-to-select/src/lib/select-container.component.ts:616]: Good catch!

**Comment by d3lm** [comment]
Nice! Thanks for this PR 🙏 Really appreciate it 💯

File: projects/ngx-drag-to-select/src/lib/select-container.spec.ts [hunk 2]
```diff
@@ -86,6 +89,22 @@ describe('SelectContainerComponent', () => {
     expect(fixture.componentInstance.selectItems.first.calculateBoundingClientRect).toHaveBeenCalled();
   });
 
+  it('should update its selection when selectable items change', (done) => {
+    selectContainerInstance.selectItems((item: SelectItemValue) => item.id === 1 || item.id === 2);
+    selectContainerInstance.itemDeselected.subscribe((item: SelectItemValue) => {
```
Comment (d3lm) [inline projects/ngx-drag-to-select/src/lib/select-container.spec.ts:94]: Newline before.

File: projects/ngx-drag-to-select/src/lib/select-container.spec.ts [hunk 3]
```diff
@@ -86,6 +89,22 @@ describe('SelectContainerComponent', () => {
...
+    });
+
+    selectContainerInstance.select.subscribe((items) => {
+      expect(items).toEqual([{ id: 2 }]);
+      done();
+    });
+
+    const previousData = testComponent.data$.value;
+    testComponent.data$.next([previousData[1], previousData[2], { id: 4 }]);
+    fixture.detectChanges();
```
Comment (d3lm) [inline projects/ngx-drag-to-select/src/lib/select-container.spec.ts:105]: Add some newlines here
```suggestion
    const previousData = testComponent.data$.value;
    
    testComponent.data$.next([previousData[1], previousData[2], { id: 4 }]);
    
    fixture.detectChanges();
```

**Review by d3lm — CHANGES_REQUESTED** [review]
Looks good! Only some minor style requests.

**Comment by lemmywrap** [comment]
Happy to contribute :)
I've added the newlines.

**Comment by lemmywrap** [comment]
Two weekends later... ;)
I'll just keep reminding you, @d3lm !

**Comment by gavinharriss** [comment]
Looks like a fix I could use as well. can't work out why the pull request isn't merged though?

**Comment by TheGlockZentix** [comment]
Would be awesome if we could get this merged @d3lm! ⭐

**Comment by d3lm** [comment]
Ok, I have done some general maintenance over the weekend and finally merged this PR. Thanks for collaborating!

**Comment by d3lm** [comment]
For your information, this is available in `v4.2.1` and it's published to npm.

**Pull request merged**